test/node/src/safe-empty-where-in.test.ts

```diff
@@ -0,0 +1,916 @@
+import {
+  destroyTest,
+  initTest,
+  TestContext,
+  testSql,
+  expect,
+  createTableWithId,
+  DIALECTS,
+  NOT_SUPPORTED,
+} from './test-setup.js'
+
+import {
+  WithSafeArrayWhereInPlugin,
+  Generated,
+  Kysely,
+  DeleteResult,
+  UpdateResult,
+} from '../../../'
+
+for (const dialect of DIALECTS) {
+  describe(`${dialect}: safe empty array where in`, () => {
+    let ctx: TestContext
+    let db: Kysely<Database>
+
+    interface Person {
+      id: Generated<number>
+      firstName: string
+    }
+
+    interface Database {
+      safeEmptyArrayPerson: Person
+    }
+
+    beforeEach(async function () {
+      ctx = await initTest(this, dialect)
+
+      db = new Kysely<Database>({
+        ...ctx.config,
+        plugins: [new WithSafeArrayWhereInPlugin()],
+      })
+
+      await db.schema.dropTable('safeEmptyArrayPerson').ifExists().execute()
+      await createTableWithId(db.schema, dialect, 'safeEmptyArrayPerson')
+        .addColumn('firstName', 'varchar(255)')
+        .execute()
+
+      await db
+        .insertInto('safeEmptyArrayPerson')
+        .values([
+          {
+            firstName: 'John',
+          },
+          {
+            firstName: 'Mary',
+          },
+          {
+            firstName: 'Tom',
+          },
+        ])
+        .execute()
+    })
+
+    describe('SELECT FROM WHERE IN TESTS', () => {
+      it('should handle empty array select from statements without throwing runtime errors', async () => {
+        const query = db
+          .selectFrom('safeEmptyArrayPerson')
+          .where('firstName', 'in', [])
+          .select('safeEmptyArrayPerson.firstName')
+
+        testSql(query, dialect, {
+          postgres: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in ($1)`,
+            ],
+            parameters: [null],
+          },
+          mysql: {
+            sql: [
+              'select `safeEmptyArrayPerson`.`firstName`',
+              'from `safeEmptyArrayPerson`',
+              'where `firstName` in (?)',
+            ],
+            parameters: [null],
+          },
+          mssql: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (@1)`,
+            ],
+            parameters: [null],
+          },
+          sqlite: {
+            sql: [
+              `select "safeEmptyArrayPerson"."firstName"`,
+              `from "safeEmptyArrayPerson"`,
+              `where "firstName" in (?)`,
+            ],
+            parameters: [null],
+          },
+        })
+
+        let result = await query.execute()
+
+        expect(result).to.deep.equal([])
```

Comment:
no need to execute really, testing the compiled sql is more than enough.

Reply:
i think we should execute to make sure there are no run-time errors, which was the case before.